project/B10732009/README.md

```diff
@@ -0,0 +1,110 @@
+# Procedural Terrain Generation #
+
+Procedurally generate random terrain with [Perlin Noise](https://en.wikipedia.org/wiki/Perlin_noise).
+
+## Basic Information ##
+
+GitHub repository : [https://github.com/B10732009/procedural-terrain-generation](https://github.com/B10732009/procedural-terrain-generation)
+
+**[Procedural Generation](https://en.wikipedia.org/wiki/Procedural_generation)** is a method of **creating data algorithmically** as opposed to manually, typically through a combination of human-generated assets and algorithms coupled with computer-generated randomness and processing power. 
+
+In video games,  **Procedural Terrain Generation** is very useful where you want to generate natural terrain (caves, hills, rivers, etc.) that has a smooth feel, but is still random.
+
+## Problem to Solve ##
+
+**Perlin Noise** is a very popular algorithm for procedural generation developed by Ken Perlin in 1983. It can be used for any kind of wave-like, undulating material, texture, or terrain. 
+
+Compared with just some random values, Perlin Noise can generate values very smoothly and continuously, which looks more realistic in terrain generation. 
+
+The graph below shows the difference between normal random values and perlin noise.
+
+<p style="text-align: center;">
+    <img src="img/compare.png" alt="compare">
+</p>
+
+In this project, I am going to implement a set of API (for C++ and Python) of the Perlin Noise algorithm in 1D, 2D, and 3D (if time is enough), and use these API to render some terrain with [Ursina](https://www.ursinaengine.org/) (a game engine in Python).
+
+## Prospective Users ##
+
+Anyone who wants to simulate random terrain or uses Perlin Noise for application.
+
+## System Architecture ##
+
+- Use `C++` to implement the algorithm.
+- Use `Pybind11` to wrap C++ functions for Python.
+- Use `Python` to render the terrain with these APIs (with [Ursina](https://www.ursinaengine.org/)).
+
+<p style="text-align: center;">
+    <img src="img/system_architecture.png" alt="system_architecture">
+</p>
+
+## API Description ##
+
+The API will have both C++ and Python version.
+
+```cpp
+// C++ API
+// get the value of a specific coordinate
+double getNoise1D(int seed=0, int octaves=1, double lacunarity=2.0, double persistance=0.5, double x);
+double getNoise2D(int seed=0, int octaves=1, double lacunarity=2.0, double persistance=0.5, double x, double y);
+double getNoise3D(int seed=0, int octaves=1, double lacunarity=2.0, double persistance=0.5, double x, double y, double z);
+
+// get the values of a specific length/area/space
+std::vector<double> getNoises1D(int seed=0, int octaves=1, double lacunarity=2.0, double persistance=0.5, double x);
+std::vector<std::vector<double>> getNoises2D(int seed=0, int octaves=1, double lacunarity=2.0, double persistance=0.5, double x, double y);
```

Comment:
Vector of vectors is a very inefficient way to store field data, although it's ok to be part of a proposal.